cmd/updater/update.sh

```diff
@@ -249,7 +255,39 @@ TEMPDIR=""
 TARFILE=""
 UPDATESRCDIR=""
 
+function download_tools_update() {
+    SPECIFIC_VERSION=$1
+    echo "downloading tools update ${SPECIFIC_VERSION}"
+    TEMPDIR=$(mktemp -d 2>/dev/null || mktemp -d -t "tmp")
+    export CLEANUP_UPDATE_TEMP_DIR=${TEMPDIR}
+    trap "rm -rf ${CLEANUP_UPDATE_TEMP_DIR}" 0
+
+    TARFILE=${TEMPDIR}/${LATEST}.tar.gz
+    UPDATESRCDIR=${TEMPDIR}/a
+    mkdir "${UPDATESRCDIR}"
+
+    if ! "${SCRIPTPATH}"/updater gettools -c "${CHANNEL}" -o "${TARFILE}" "${BUCKET}" "${SPECIFIC_VERSION}"
+    then
+        echo Error downloading tools tarfile
```

Comment:
missing quotes around these echo strings. Looks like the rest of this file is a bit inconsistent as well, but we may as well default to the convention we use in most of our scripts.

Reply:
Added quotes around echo strings.  

> This looks almost correct, but I'm not sure how useful it is without also restarting the dispenser. You still have a manual process that you have to remember every time.
> 
> Why not put a simple script together to drop in the crontab ahead of `update.sh` which does exactly what is needed?
> 
> ./update_dispenser.sh beta algorand-releases /path/to/algod/binaries/ /path/to/dispenser/
> 
> ```
> CHANNEL=$1
> BUCKET=$2
> BINDIR=$3
> DISPENSER_DIR=$4
> 
> CURRENTVER="$(( ${BINDIR}/algod -v 2>/dev/null || echo 0 ) | head -n 1)"
> LATEST="$(${SCRIPTPATH}/updater ver check -c ${CHANNEL} ${BUCKET} | sed -n '2 p')"
> if [ ${CURRENTVER} -ge ${LATEST} ]; then
>     return 1
> fi
> 
> ${BINDIR}/updater gettools -c "${CHANNEL}" -o "/tmp/tools.tar.gz" "${BUCKET}"
> tar xvf /tmp/tools.tar.gz -o "${DISPENSER_DIR}"
> 
> pkill dispenser
> ${DISPENSER_DIR}/dispenser
> ```

This does not support all of the options of the updater.sh script, and fetching the tools may be generally useful for things besides the dispenser.   Also, the dispenser runs as a service, which needs to be stopped prior to updating the binary.  I am planning to put this logic in a separate crontab script which will check for a new dispenser binary in the tools directory.